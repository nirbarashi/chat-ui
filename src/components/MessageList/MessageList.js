import React from 'react';
import "./MessageList.scss";
import Message from "../Message/Message";

const MessageList = () => {
	return (
		<main className="chat-messages">
			<ol className="chat-messages-list">
				<Message />
				<Message isSelf={true}/>
			</ol>
		</main>
	)
}

export default MessageList;