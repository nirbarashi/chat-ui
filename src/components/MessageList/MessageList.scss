.chat-messages{
  flex:1; background-color:#e5ddd5; padding:20px; overflow-y: auto;
  &-list{display:flex; flex-direction:column;}
}