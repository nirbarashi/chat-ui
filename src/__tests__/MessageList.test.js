import React from "react";
import { create } from "react-test-renderer";
import MessageList from "../components/MessageList/MessageList";


describe("MessageList component", () => {
	test("Matches the snapshot", () => {
		const ml = create(<MessageList messages={[{ username: 'Nir', text: 'Welcome friend, feel free to send messages ', avatar: 'pikachu'}, { username: 'Alon22', text: 'Thank you !', avatar: 'snorlax'}]} username="Nir" />);
		expect(ml.toJSON()).toMatchSnapshot();
	});
});