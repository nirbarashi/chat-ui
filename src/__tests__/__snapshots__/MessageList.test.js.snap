// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList component Matches the snapshot 1`] = `
<main
  className="chat-messages"
>
  <ol
    className="chat-messages-list"
  >
    <li
      className="chat-message is-self"
    >
      <header
        className="profile"
      >
        <img
          alt=""
          className="profile-image"
          src="003-pikachu.png"
        />
        <span
          className="profile-name"
        >
          Nir
        </span>
      </header>
      <p
        className="chat-message-content"
      >
        Welcome friend, feel free to send messages 
      </p>
    </li>
    <li
      className="chat-message "
    >
      <header
        className="profile"
      >
        <img
          alt=""
          className="profile-image"
          src="001-snorlax.png"
        />
        <span
          className="profile-name"
        >
          Alon22
        </span>
      </header>
      <p
        className="chat-message-content"
      >
        Thank you !
      </p>
    </li>
  </ol>
</main>
`;
